cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -494,13 +581,21 @@ public Set<RunId> restart(ApplicationId applicationId, long startTimeSeconds, lo
     return runs;
   }
 
+  public Set<RunId> restart(NamespaceId namespaceId, String appName, long startTimeSeconds, long endTimeSeconds)
+    throws Exception {
+    ApplicationId applicationId = getLatestApplicationId(namespaceId, appName);
+    return restart(applicationId, startTimeSeconds, endTimeSeconds);
+  }
+
   /**
-   * Stop all active programs for the given application
+   * Stop all active programs for the given application across all versions
```

Comment:
this means store.getActiveRuns(namespace, appname) returns all active runs across all versions? If so it would be more clear to rename it getAllActiveRuns(), as most of the other methods that take application reference will return things just for the latest version.